### ISSUE TITLE
Redfish: Code Update - Enhance redfishtool to support UpdateService

### ISSUE THREAD

**Issue by geissonator** [ibm-openbmc/dev#500]
This is to track adding firmware update support to the redfish tool.

I've started initial support out at https://github.com/geissonator/Redfishtool/tree/addUpdateServiceSupport

https://github.com/DMTF/Redfishtool/issues/62 tracks the feature upstream.

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by geissonator** [comment]
Code is done - PR up at https://github.com/DMTF/Redfishtool/pull/63

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by geissonator** [comment]
DMTF has stated that because the current UpdateService is still considered OEM, they will not take my changes as-is.  https://github.com/ibm-openbmc/dev/issues/483 is where the new "official" implementation will occur so until that is in, we can not proceed on this story. Moving it under our 2.8 release.

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by rfrandse** [comment]
refresh

**Comment by lxwinspur** [comment]
@geissonator @mzipse 
Around this issue, Do we need to wait for #1732 to close before proceeding?
AFAIK, in Ed's task list, #1732 has a very low priority.
https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/32174

**Comment by geissonator** [comment]
> @geissonator @mzipse
> Around this issue, Do we need to wait for #1732 to close before proceeding?
> AFAIK, in Ed's task list, #1732 has a very low priority.
> https://gerrit.openbmc-project.xyz/c/openbmc/bmcweb/+/32174

Yeah, I think we should have the back end implementation done so that we can fully test the work done within this story to utilize it.

**Comment by gtmills** [comment]
https://gerrit.openbmc.org/c/openbmc/bmcweb/+/32174 merged, this work could be done at https://github.com/DMTF/Redfishtool
